### PR TITLE
MAISTRA-2613 Update to latest Istio 1.9.8

### PR DESCRIPTION
**Pull request by shamsher31**

### CODE REVIEW

**Comment by shamsher31** [comment]
@jwendell There was a conflict in istio.deps for proxy sha as you mentioned, so I have left it as it as we have a seperate proxy for maistra. because of which few tests are still failing.

**Comment by shamsher31** [comment]
I think we need to update something once https://github.com/maistra/envoy/pull/104 is merged?

File: go.mod
Comment (shamsher31) [inline go.mod:110]: Is this also needs to be updated to https://github.com/istio/client-go/releases/tag/1.9.8?

I could not find any commit related to upgrading it to 1.9.8.

Comment (jwendell) [reply go.mod:110]: don't worry about this right now. This can be updated by running `bin/update_deps.sh` later, after this is merged

**Comment by jwendell** [comment]
> I think we need to update something once [maistra/envoy#104](https://github.com/maistra/envoy/pull/104) is merged?

nope. After this Envoy PR merges, automator will take care of the rest.

**Comment by jwendell** [comment]
@shamsher31 All tests pass with the latest (1.9.8) proxy: https://github.com/maistra/istio/pull/429
Feel free to update istio.deps in this PR... Or get the PR above merged first (needs a second approval)...

**Comment by shamsher31** [comment]
@jwendell Test failures are now fixed. PTAL.
I think there are only a few changes in charts, so the operator side will be quick to update.

**Comment by luksa** [comment]
LGTM 

Do we want to merge this via a merge-commit so the changes are nicely separated from the maistra commits?

**Review by luksa — APPROVED** [review]

**Comment by jwendell** [comment]
> Do we want to merge this via a merge-commit so the changes are nicely separated from the maistra commits?

@luksa What do you mean? I already added the `not squash` label. Isn't it enough?

File: tools/istio-iptables/pkg/cmd/run_test.go
Comment (jwendell) [inline tools/istio-iptables/pkg/cmd/run_test.go:421]: why?
Why does it pass upstream and fail with us?

File: pilot/pkg/networking/core/v1alpha3/gateway_test.go
Comment (jwendell) [inline pilot/pkg/networking/core/v1alpha3/gateway_test.go]: why this commit was necessary?
I mean, this commit essentially makes this file equal to upstream. Does it mean we changed it before? Maybe in `fix test` commit? If so, it should be enough to remove it, right?

File: pkg/servicemesh/federation/status/manager_test.go
Comment (jwendell) [inline pkg/servicemesh/federation/status/manager_test.go]: I **think** this should be
```
if len(status.DiscoveryStatus.Inactive) == 0 || (len(status.DiscoveryStatus.Inactive) > 0 && status.DiscoveryStatus.Inactive[0].Watch.LastEvent.IsZero())
```

@rcernich what's your input here?

**Comment by luksa** [comment]
Even with don't squash, we end up with a linear history. As much as I prefer linear history for day-to-day stuff, I think that in this particular case a merge commit would be better. This is what we'd end up with:
```
       U1---U2---U3 
     /              \
1.9.5---M1--M2--M3---X  maistra-2.1
```
where `U1..U3` are upstream commits (from 1.9.5. to 1.9.8), `M1-M3` are the existing Maistra commits that we added on top of 1.9.5, and `X` is the merge commit that merges the two linear histories. 

This will leave the resulting history slightly better organized, as it would keep our changes separate from those upstream.

File: tools/istio-iptables/pkg/cmd/run_test.go
Comment (shamsher31) [reply tools/istio-iptables/pkg/cmd/run_test.go:421]: I compared the test output and saw that the actual output needs `-w`. not sure what it does.

**Comment by jwendell** [comment]
Ah, got it. We don't use merge strategy by default. Let me look if there's an option for that

File: pilot/pkg/networking/core/v1alpha3/gateway_test.go
Comment (shamsher31) [reply pilot/pkg/networking/core/v1alpha3/gateway_test.go]: Not sure if we added it before. The test was failing because [tlsParams](https://github.com/maistra/istio/pull/430/commits/a2dad5095041f24e3ee0fa4d095e29f664f4b9f0#diff-7abd8e9bc7226623690e85b5acc0280def2d72df7960b1324349b08dcb186ca1L458) was removed from upstream, which makes this option obsolete for most of the test. I tried to just pass empty params, but it wasn't used in the test. so lint throws a warning.

**Comment by jwendell** [comment]
https://github.com/maistra/test-infra/pull/172

File: tools/istio-iptables/pkg/cmd/run_test.go
Comment (shamsher31) [reply tools/istio-iptables/pkg/cmd/run_test.go:421]: I think it has something to do with the output of the following?
https://github.com/maistra/istio/blob/9c6658fd6b16846d6cff4a525585a1dcb949f2a6/tools/istio-iptables/pkg/cmd/run_test.go#L58

Comment (jwendell) [reply tools/istio-iptables/pkg/cmd/run_test.go:421]: This was likely added by https://github.com/maistra/istio/pull/242
@luksa Any input here?

Comment (rcernich) [reply tools/istio-iptables/pkg/cmd/run_test.go:421]: I believe that was added to keep it from failing if it was trying to update iptables at the same time as the platform.

Comment (jwendell) [reply tools/istio-iptables/pkg/cmd/run_test.go:421]: True. My question was more like, Has this changed upstream between 1.9.5 and 1.9.8? Because those tests are passing currently and failing with the rebase.

Comment (rcernich) [reply tools/istio-iptables/pkg/cmd/run_test.go:421]: Looking at these changes, I think the test may have changed.  I'd check the gid and uid that are being set.  Should be easy to see from the test failures.  The only thing we do differently is allow uid and gid to be dynamically specified, as opposed to hard-coded (1337 is the default).

Comment (luksa) [reply tools/istio-iptables/pkg/cmd/run_test.go:421]: We add the `-w`, but upstream does not.

File: pkg/servicemesh/federation/status/manager_test.go
Comment (rcernich) [reply pkg/servicemesh/federation/status/manager_test.go]: Seems like `if len(status.DiscoveryStatus.Inactive) == 0 || status.DiscoveryStatus.Inactive[0].Watch.LastEvent.IsZero()` is enough.  I don't remember what that is doing, but I suspect if this panics, that's just a test failure, so all we're doing by cleaning things up is making sure the test doesn't crash?

**Review by jwendell — DISMISSED** [review]

File: pilot/pkg/networking/core/v1alpha3/gateway.go
Comment (dgn) [inline pilot/pkg/networking/core/v1alpha3/gateway.go]: this is removing an OSSM feature (setting TLS defaults through env vars), we need this

**Comment by dgn** [comment]
if we want to use merge strategy here, let's keep the required changes on top of the upstream commits to a single commit - we currently have 6 additional commits in here

**Comment by shamsher31** [comment]
> if we want to use merge strategy here, let's keep the required changes on top of the upstream commits to a single commit - we currently have 6 additional commits in here

Thanks, I have squashed them into a single commit.

File: pilot/pkg/networking/core/v1alpha3/gateway.go
Comment (shamsher31) [reply pilot/pkg/networking/core/v1alpha3/gateway.go]: Reverted, Thanks.

File: pilot/pkg/networking/core/v1alpha3/gateway.go
Comment (shamsher31) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:517]: This was added in https://github.com/maistra/istio/commit/443879e3cb83b724b6cbb0ab582b8bda13742e94#diff-7abd8e9bc7226623690e85b5acc0280def2d72df7960b1324349b08dcb186ca1R492

**Comment by shamsher31** [comment]
Integration tests are failing due to timeout. It's almost taking more than 1hr to run the integration test.
```
traffic.go:150: call failed from TestRunner to https://172.18.255.155:443 (using https): 
timeout while waiting (last error: no responses received)
```

**Review by dgn — APPROVED** [review]

**Pull request merged**
